Sc analysis

andrewram4287 opened the pull request.

aarmey reviewed (changes requested):
What does this do? What does it aim to do? Please describe.

andrewram4287 commented:
> What does this do? What does it aim to do? Please describe.

Uploaded all the PopAlign data, filtered genes, performed NNMF, converted from DF to Xarray, ran minimization of loglik for the given data to determine the factors, and also ran cross validation to find the best factor/cluster combination 

Also, added more descrioption to each function

aarmey commented on gmm/figures/figure7.py:
```diff
@@ -1,34 +1,122 @@
+#     print(repr(x.device_buffer.device()))  # CpuDevice(id=0)
+
+
+def geneGMM_CV(X, numFolds: int, numClusters: int, numRank: int, maxiter=300):
```
Is there a reason you're not able to use the existing `tensorGMM_CV()`?

andrewram4287 replied:
Nope! Just fixed it!

aarmey commented on gmm/figures/figure7.py:
```diff
@@ -1,34 +1,122 @@
+# def jax_run():
+#     """Allows JAX to run on GPU"""
+#     # Global flag to set a specific platform, must be used at startup
+#     jax.config.update('jax_platform_name', 'cpu')
```
You can just leave this one line commented out at the top of the file. No need for a whole function.

aarmey reviewed (changes requested):
A couple small adjustments.

aarmey commented on gmm/figures/figure7.py:
```diff
@@ -1,34 +1,100 @@
+def gene_import(offset):
+    """Imports gene data from PopAlign and perfroms gene filtering process"""
+    genesDF, geneNames = import_thompson_drug()
+    genesN = normalizeGenes(genesDF, geneNames)
```
@andrewram4287 last question—you don't seem to ever use `genesN` here. Is that intended?

andrewram4287 replied:
Shoot that is not intended. `mu_sigma(genesDF,geneNames) ` should be `mu_sigma(genesN,geneNames) `. I will run that again and get the correct csv data uploaded with the normalization included.
